baybe/constraints/validation.py

```diff
@@ -26,6 +32,8 @@ def validate_constraints(  # noqa: DOC101, DOC103
         ValueError: If any discrete constraint includes a continuous parameter.
         ValueError: If any discrete constraint that is valid only for numerical
             discrete parameters includes non-numerical discrete parameters.
+        ValueError: If any parameter affected by a cardinality constraint does
```

Comment:
I'm not entirely sure if I understand your question here. But the rationale is that we disallow any parameter whose range does not cover 0 to be part of a cardinality constraint since that parameter is anyway unable to affect the constraint – it will always be "active". And hence, you can always achieve the same result by removing that parameter from the constraint and reducing the number of allowed active values of the respective constraint by 1. Does it make sense?

Reply:
Then it seems that I did not realize how such parameters are being handled. Will keep this comment open as a reminder for me to verify this.

Reply:
Resolving this for now as this will get a fresh review from me after the huge changes.